plugin/pkg/shoot/validator/admission.go

```diff
@@ -591,7 +598,13 @@ func (c *validationContext) ensureMachineImages() field.ErrorList {
 	if c.shoot.DeletionTimestamp == nil {
 		for idx, worker := range c.shoot.Spec.Provider.Workers {
 			fldPath := field.NewPath("spec", "provider", "workers").Index(idx)
-			image, err := ensureMachineImage(c.oldShoot.Spec.Provider.Workers, worker, c.cloudProfile.Spec.MachineImages, fldPath)
+			coreCloudProfile := core.CloudProfile{}
+			if err := gardencorev1beta1.Convert_v1beta1_CloudProfile_To_core_CloudProfile(c.cloudProfile, &coreCloudProfile, nil); err != nil {
+				allErrs = append(allErrs, field.InternalError(fldPath, err))
+				return allErrs
+			}
```

Comment:
In the quota validator, you convert the core shoot to versioned shoot, why not be consistent and do the same here instead of converting the cloudprofile?